File: src/main/java/org/app/infrastructure/local/UserData.java
package org.app.infrastructure.local;

import org.app.domain.User;

import java.util.Date;
import java.util.UUID;

public class UserData{
    private final String id;
    private final String username;
    private final Date createdAt;

    public UserData(String id, String username) {
        this.id = id;
        this.username = username;
        this.createdAt = new Date();
    }

    public String getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public static UserData toData(User user){
        return new UserData(user.getId().toString(), user.getUsername());
    }

    public static User toDomain(UserData userData){
        return new User(UUID.fromString(userData.id), userData.username);
    }
}

File: go/app/main/infrastructure/local/user_data.go
package local

import (
    "github.com/google/uuid"
    "mutualAttentionSystem/app/main/domain"
    "time"
)

type UserData struct {
    ID        string
    Username  string
    CreatedAt time.Time
}

func toDomain(userData *UserData) *domain.User {
    var relationship []*domain.Relationship
    return &domain.User{ID: uuid.MustParse(userData.ID), Username: userData.Username, Relationships: relationship}
}

func toData(user *domain.User) *UserData {
    return &UserData{ID: user.ID.String(), Username: user.Username, CreatedAt: time.Now()}
}
